### PR TITLE
Support incoming unscaled bptPrice

### PR DESCRIPTION
**Pull request by EndymionJkb**
Assume the incoming bptPrice was calculated with an unscaled token balance. Since our calculations assume all balances are scaled, we must adjust it.

### CODE REVIEW

File: pkg/pool-weighted/test/ManagedPoolSettings.test.ts [hunk 1]
```diff
@@ -826,7 +826,13 @@ describe('ManagedPoolSettings', function () {
   });
 
   describe('circuit breakers', () => {
-    async function getBptPrice(): Promise<BigNumber> {
+    async function getUnscaledBptPrice(): Promise<BigNumber> {
+      const totalSupply = await pool.getActualSupply();
+
+      return fpDiv(fpMul(totalSupply, poolWeights[0]), initialBalances[0]);
```
Comment (nventuro) [inline pkg/pool-weighted/test/ManagedPoolSettings.test.ts:832]: The first token has 18 decimals, so we're actually not testing anything

File: pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol [hunk 2]
```diff
@@ -1027,14 +1027,21 @@ abstract contract ManagedPoolSettings is BasePool, ProtocolFeeCache, IControlled
...
+
+        _circuitBreakerState[token] = CircuitBreakerStorageLib.setCircuitBreaker(
+            scaledBptPrice,
             normalizedWeight.complement(),
             lowerBoundPercentage,
             upperBoundPercentage
         );
 
-        emit CircuitBreakerSet(token, bptPrice, lowerBoundPercentage, upperBoundPercentage);
+        emit CircuitBreakerSet(token, scaledBptPrice, lowerBoundPercentage, upperBoundPercentage);
```
Comment (nventuro) [inline pkg/pool-weighted/contracts/managed/ManagedPoolSettings.sol:1044]: I think it'd be good to have both the getter and this event return the unscaled bpt price, to make things easier for the outside world

File: pkg/pool-weighted/test/ManagedPoolSettings.test.ts (hunk 1 above)
Comment (EndymionJkb) [reply pkg/pool-weighted/test/ManagedPoolSettings.test.ts:832]: For now, making it use a dynamic tokenIndex.

File: pkg/pool-weighted/test/ManagedPoolSettings.test.ts [hunk 3]
```diff
@@ -951,15 +969,12 @@ describe('ManagedPoolSettings', function () {
...
+            const expectedWeightComplement = FP_ONE.sub(poolWeights[tokenIndex]);
             const totalSupply = await pool.getActualSupply();
-            const scalingFactors = await pool.getScalingFactors();
 
-            const expectedBptPrice = fpDiv(
-              fpMul(totalSupply, poolWeights[0]),
-              fpMul(initialBalances[0], scalingFactors[0])
-            );
+            // Don't scale; the getter will return the unscaled value.
+            const expectedBptPrice = fpDiv(fpMul(totalSupply, poolWeights[tokenIndex]), initialBalances[tokenIndex]);
```
Comment (nventuro) [inline pkg/pool-weighted/test/ManagedPoolSettings.test.ts:977]: Shouldn't this test that the pool returns the bpt price that we set?

Comment (nventuro) [reply pkg/pool-weighted/test/ManagedPoolSettings.test.ts:977]: I guess the computation here returns the same thing. We could replace it with `getUnscaledBptPrice()` though

**Review by nventuro — APPROVED** [review]

**Pull request merged**